app/scripts/controllers/transactions/index.js

```diff
@@ -336,6 +337,268 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
+  // ====================================================================================================================================================
+
+  /**
+   * @param {number} txId
+   * @returns {TransactionMeta} the txMeta who matches the given id if none found
+   * for the network returns undefined
+   */
+  _getTransaction(txId) {
+    const { transactions } = this.store.getState();
+    return transactions[txId];
+  }
+
+  _checkIfTxStatusIsUnapproved(txId) {
+    return (
+      this.txStateManager.getTransaction(txId).status ===
+      TRANSACTION_STATUSES.UNAPPROVED
+    );
+  }
+
+  _updateTransaction(txId, proposedUpdate, note) {
+    const txMeta = this.txStateManager.getTransaction(txId);
+    const updated = merge(txMeta, proposedUpdate);
+    this.txStateManager.updateTransaction(updated, note);
+  }
+
+  /**
+   *
+   * @param {string} txId - transaction id
+   * @param {object} editableParams - holds the eip1559 fees parameters
+   * @param editableParams.data
+   * @param editableParams.from
+   * @param editableParams.to
+   * @param editableParams.value
+   * @param editableParams.gas
+   * @param editableParams.gasPrice
+   */
+  updateEditableParams(txId, { data, from, to, value, gas, gasPrice }) {
```

Comment:
I know we agreed upon this name in another thread. If we feel good about this I 'm not gonna block on it. However, all params are "editable"

Also gasPrice is not a part of this when we have explicit gas update methods? I'm confused.

Reply:
I added the gasPrice because of [this comment](https://github.com/MetaMask/metamask-extension/pull/13496#discussion_r812346125).

Thinking about it, now I can split this into two calls. Where the second call will use the dedicated gasUpdate method.

Reply:
Fixed [here](https://github.com/MetaMask/metamask-extension/pull/13652/commits/c0c75ea0e1df26d077ce5b1186d2fc8462b79d9d)